### DISCUSSION TITLE
Can I use other modules?

### DISCUSSION THREAD

**Discussion by Terminou** [exodrifter/unity-python#30] [Q&A]
I have a Python program that makes object classification. I have imported the following in order to use the pre-trained model.

```
import numpy as np
import cv2
import pickle
```

I want to use this model on Unity's camera frames. Is it possible? How?

**Comment by exodrifter** [comment]
This library depends on IronPython, and as such it cannot use libraries like `numpy` or `cv2` as addressed in #22.

Disregarding that, I would think it is possible -- you can write the camera output to a texture and then send that to your python code using a variable. However, I don't believe there's an object classification library out there that is supported by IronPython (If I'm wrong about that, please do let me know.). So, I think you're out of luck there.

**Comment by Terminou** [comment]
I handled my problem using the ZeroMQ approach. I encoded the string image to base 64 and then sent it to Python. In order to open the image using OpenCV I decoded it and it worked well.

![unnamed](https://user-images.githubusercontent.com/27548058/85856360-8e772c00-b7c0-11ea-8b15-e08e6342d441.gif)

**Comment by exodrifter** [comment]
Awesome! Glad you got it working!
